Document and rename overload-overlap error code (#16074)
python/mypy: docs/source/more_types.rst

``str``. Both conditions are true, so mypy will correctly flag ``unsafe_func`` as
being unsafe.

> Note that in cases you ignore the overlapping overload error, mypy will usually

Reviewer: ```suggestion
Note that if you ignore the overlapping overload error, mypy will usually
```
